### DISCUSSION TITLE
Get the total pages

### DISCUSSION THREAD

**Discussion by akung-at-julo** [MatthiasValvekens/pyHanko#267] [Q&A] [answered]
Hello, this package is so awesome, although I still using the alpha version.

I frequently get the error `PdfError: Page index out of range`, because the signature page is more than available.
How can I get the total pages of the pdf file? So I can put the signature on the correct page? 

Thank you

**Comment by peteris-zealid** [comment]
If you want the signature to be on the last page you can specify `-1` as the page number. Hope this helps.

**Reply by akung-at-julo** [reply]
nice trick, will try it. 

But, can I get the total pages, so I can add more logic to the page?

**Reply by peteris-zealid** [reply]
I remember seeing it when reading though the code. I think the variable is called `pages` and you can try to get the `len` of that. I am not a maintainer, but just a fellow user, so this is as much as I can help.

**Reply by akung-at-julo** [reply]
thank you @peteris-zealid for the clue, will search for it.

**Comment by MatthiasValvekens** [comment, marked as answer]
Hi @akung-at-julo,

I don't remember offhand if it's exposed conveniently in the pyHanko API, but in a well-formed PDF, the **Count** entry on the page tree root gives the full page count of the document. So you can always do something like this:

```python

r = PdfFileReader(...)
page_count = int(r.root['/Pages']['/Count'])
```

**Reply by akung-at-julo** [reply]
cool this works
![image](https://github.com/MatthiasValvekens/pyHanko/assets/130727237/a322c618-dc76-43f8-b434-7e09c8778a9e)
